### ISSUE TITLE
[Bug]: Warning:  Unknown: [ddappsec] Error sending message for command config_sync: dd_network in Unknown on line 0

### ISSUE THREAD

**Issue by Skaopap** [DataDog/dd-trace-php#2257]
### Bug report

When the api B is calling the api A, the string "Unknown: [ddappsec] Error sending message for command config_sync: dd_network in Unknown on line 0" is sometimes added into its response, resulting in a parsing error from the api B that is expecting a json.

Error displayed by the api B receiving response from api A : 

```
Received a response that is invalid JSON: <br />
<b>Warning</b>:  Unknown: [ddappsec] Error sending message for command config_sync: dd_network in <b>Unknown</b> on line <b>0</b><br />
{"data":XXX}
```

![image](https://github.com/DataDog/dd-trace-php/assets/36385604/aa412e07-ee3e-4ec5-a441-8eb4e82ae140)

After some digging I really don't know where it's coming from and how to fix it, especially since ASM is not enabled 😥

Info displayed below is from api A.

### PHP version

PHP 8.1.21 (cli) (built: Jul 28 2023 12:58:04) (NTS) Copyright (c) The PHP Group Zend Engine v4.1.21, Copyright (c) Zend Technologies  
  with Zend OPcache v8.1.21, Copyright (c), by Zend Technologies       
  with ddtrace v0.90.0, Copyright Datadog, by Datadog       
  with ddappsec v0.10.0, Copyright Datadog, by Datadog  

### Tracer or profiler version

0.90.0

### Installed extensions

[PHP Modules]
apcu
Core
ctype
curl
date
ddappsec
ddtrace
dom
fileinfo
filter
ftp
hash
iconv
intl
json
libxml
mbstring
mysqlnd
openssl
pcre
PDO
pdo_mysql
pdo_sqlite
Phar
posix
readline
Reflection
session
SimpleXML
sodium
SPL
sqlite3
standard
tokenizer
xml
xmlreader
xmlwriter
Zend OPcache
zip
zlib

[Zend Modules]
Zend OPcache
ddappsec
ddtrace

### Output of `phpinfo()`

{
    "date": "2023-09-14T09:40:16Z",
    "os_name": "Linux XXX",
    "os_version": "5.10.186-179.751.amzn2.x86_64",
    "version": "0.90.0",
    "lang": "php",
    "lang_version": "8.1.15",
    "env": "prod",
    "enabled": true,
    "service": "my-api",
    "enabled_cli": false,
    "agent_url": "http:\/\/xx.xx.xx.xx:8126",
    "debug": false,
    "analytics_enabled": false,
    "sample_rate": 1,
    "sampling_rules": [],
    "tags": [],
    "service_mapping": [],
    "distributed_tracing_enabled": true,
    "priority_sampling_enabled": true,
    "dd_version": 1.0",
    "architecture": "x86_64",
    "sapi": "cli",
    "datadog.trace.request_init_hook": "\/opt\/datadog\/dd-library\/0.90.0\/dd-trace-sources\/bridge\/dd_wrap_autoloader.php",
    "open_basedir_configured": false,
    "uri_fragment_regex": null,
    "uri_mapping_incoming": null,
    "uri_mapping_outgoing": null,
    "auto_flush_enabled": false,
    "generate_root_span": true,
    "http_client_split_by_domain": false,
    "measure_compile_time": true,
    "report_hostname_on_root_span": false,
    "traced_internal_functions": null,
    "auto_prepend_file_configured": false,
    "integrations_disabled": "default",
    "enabled_from_env": false,
    "opcache.file_cache": null
}

**Comment by Anilm3** [comment]
@Skaopap if you haven't enabled ASM, the extension will still try to launch the external daemon to allow you to enable it through remote configuration if you so wish. This error suggests a failure communicating with said daemon which could be for many reasons, to get rid of this error there are two options:

- Explicitly disabling ASM by setting `datadog.appsec.enabled = Off` in the relevant ini file (e.g. `98-ddtrace.ini`), or using the corresponding environment variable: `DD_APPSEC_ENABLED=false`
- Finding the root cause for the communication issue, in which case we would have to assist you through a support ticket as we might need logs for that.

cc: @estringana

**Comment by Skaopap** [comment]
I've already tried in a development environment to explicitly disable ASM by setting DD_APPSEC_ENABLED=false, but as ddappsec was still displayed in the output of php -v, I told myself this wouldn't solve the problem.

After seeing your article, I added this environment variable again in an environment that had this problem, and it did indeed solve it. 

Thank you very much! 😁

I guess we'll now always disable ASM explicitly to avoid this problem again.
